fastestimator/estimator/trace.py

```diff
@@ -195,6 +195,97 @@ def on_epoch_end(self, state):
 
     def on_end(self, state):
         state['total_time'] = "{} sec".format(round(time.perf_counter() - self.train_start, 2))
+        for model_name, model in self.network.model.items():
+            state[model_name + "_lr"] = round(backend.get_value(model.optimizer.lr), 6)
+
+
+class LRController(Trace):
+    def __init__(self,
+                 model_name,
+                 lr_schedule=None,
+                 reduce_on_eval=False,
+                 reduce_patience=10,
+                 reduce_factor=0.1,
+                 reduce_mode="min",
+                 min_lr=1e-6):
+        if reduce_on_eval and isinstance(reduce_on_eval, str):
```

Comment:
this can just be 
`if is instance(reduce_on_eval, str)`

Reply:
or even get rid of the outer if and do

super().__init__(inputs=reduce_on_eval if isinstance(reduce_on_eval,str) else None)

Reply:
fixed